### PR TITLE
feat(api-client):  server variables from example

### PR DESCRIPTION
**Pull request by antlio**
this pr retrieves the server variables from an open api document in order to display them in the request path variables tables as seen below:

image

### CODE REVIEW

File: packages/api-client/src/components/CodeInput/codeVariableWidget.ts [hunk 1]
```diff
@@ -165,7 +165,7 @@ export const pillPlugin = (props: {
 
         for (const { from, to } of view.visibleRanges) {
           const text = view.state.doc.sliceString(from, to)
-          const regex = /{{(.*?)}}/g
+          const regex = /{{((?:[^{}]|{[^{}]*})*)}}/g
```
[the remark this answers is not in the document]
Comment (geoffgscott) [reply packages/api-client/src/components/CodeInput/codeVariableWidget.ts:168]: Can we please put this one into a helper with some unit testing?

**Comment by antlio** [comment]
as discussed with @tmastrom one check is failing due to an issue with galaxy.scalar.com but will be fix once migration is done

File: playwright/tests/testPlayButton.ts [hunk 2]
```diff
@@ -15,7 +15,9 @@ export async function testPlayButton(page: Page) {
   await page.waitForSelector('.scalar-client')
 
   // URL
-  await expect(page.getByText('https://galaxy.scalar.com')).toBeVisible()
+  await expect(
+    page.locator('.cm-pill', { hasText: 'https://galaxy.scalar.com' }),
+  ).toBeVisible()
```
Comment (tmastrom) [inline playwright/tests/testPlayButton.ts:20]: Looks like we had some changed elements. Nice update

**Comment by hanspagel** [comment]
Let’s give #2931 priority, so we don’t cause more conflicts.

**Review by marclave — APPROVED** [review]

**Comment by marclave** [comment]
i think we can eventually do something with servers, but path is a good place to start!

**Review by marclave — APPROVED** [review]

**Pull request merged**